/v1/regard-device-attribute-value-change - value not updating in MWDI cache

When a device attribute value change is performed, the value is not getting updated in MWDI cache.

Build: 1.1.2j

API : /v1/regard-device-attribute-value-change

Response: 204 OK but value not reflecting MWDI cache

Body:
{"notification-proxy-1-0:attribute-value-changed-notification":{"counter":6,"timestamp":"2025-02-13T08:01:28.0+00:00","object-path":"/core-model-1-4:network-control-domain=live/control-construct=513250006/logical-termination-point=LTP-MWPS-TTP-ODU-A/layer-protocol=LP-MWPS-TTP-ODU-A/air-interface-2-0:air-interface-pac/air-interface-configuration","attribute-name":"atpc-is-on","new-value":"false"}}

{"notification-proxy-1-0:attribute-value-changed-notification":{"counter":6,"timestamp":"2025-02-13T08:01:28.0+00:00","object-path":"/core-model-1-4:network-control-domain=live/control-construct=513250006/logical-termination-point=LTP-MWPS-TTP-ODU-A/layer-protocol=LP-MWPS-TTP-ODU-A/air-interface-2-0:air-interface-pac/air-interface-configuration","attribute-name":"atpc-is-on","new-value":"false"}}

```
Docker logs:

[13.02.2025 07:52.30.262] [LOG]   Send data to Requestor:http://172.28.127.18:3008/core-model-1-4:network-control-domain=live/control-construct=513250006/logical-termination-point=LTP-MWPS-TTP-ODU-A/layer-protocol=LP-MWPS-TTP-ODU-A/air-interface-2-0:air-interface-pac/air-interface-configuration
[13.02.2025 07:52.31.402] [LOG]
 callback : http://172.28.127.19:8181/rests/data/network-topology:network-topology/topology=topology-netconf/node=513250006/yang-ext:mount/core-model-1-4:control-construct/logical-termination-point=LTP-MWPS-TTP-ODU-A/layer-protocol=LP-MWPS-TTP-ODU-A/air-interface-2-0:air-interface-pac/air-interface-configuration header :{"Authorization":"Basic bXdkaV90ZXN0YTptd2RpQDEyMw=="}body :""response code:200
[13.02.2025 07:52.31.424] [ERROR] ReferenceError: indice is not defined
    at assignValueToJson (/home/openbackhaul/microWaveDeviceInventory/service/individualServices/cacheUpdateBuilder.js:169:34)
    at Object.exports.cacheUpdateBuilder (/home/openbackhaul/microWaveDeviceInventory/service/individualServices/cacheUpdateBuilder.js:80:5)
    at /home/openbackhaul/microWaveDeviceInventory/service/IndividualServicesService.js:4991:41
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
GET /core-model-1-4:network-control-domain=live/control-construct=513250006/logical-termination-point=LTP-MWPS-TTP-ODU-A/layer-protocol=LP-MWPS-TTP-ODU-A/air-interface-2-0:air-interface-pac/air-interface-configuration 200 1172.991 ms - 16660
POST /v1/regard-device-attribute-value-change 204 1243.403 ms - -
[13.02.2025 07:52.31.545] [ERROR] Error: connect ECONNREFUSED 127.0.0.1:4013
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 4013,
  config: {
    url: 'HTTP://127.0.0.1:4013/v1/regard-attribute-value-change',
    method: 'post',
    data: '{"MicroWaveDeviceInventory-1-2:attribute-value-changed-notification":{"counter":4,"timestamp":"2025-02-13T07:52:30.0+00:00","attribute-name":"atpc-is-on","new-value":"true"}}',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      'x-correlator': 'e5F26ad2-dDF8-64Df-F178-7C3EEd0E810c',
      'trace-indicator': '1',
      user: 'MicroWaveDeviceInventory',
      originator: 'MicroWaveDeviceInventory',
      'customer-journey': 'unknown',
      'operation-key': 'Operation key not yet provided.',
      'User-Agent': 'axios/0.21.4',
      'Content-Length': 174
    },
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    adapter: [Function: httpAdapter],
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    validateStatus: [Function: validateStatus],
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    }
  },
  request: <ref *1> Writable {
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: true,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      buffered: [],
      bufferedIndex: 0,
      allBuffers: true,
      allNoop: true,
      pendingcb: 0,
      constructed: true,
      prefinished: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: true,
      errored: null,
      closed: false,
      closeEmitted: false,
      [Symbol(kOnFinished)]: []
    },
    _events: [Object: null prototype] {
[13.02.2025 07:52.31.542] [LOG]   starting notification of 1 subscribers for '/v1/notify-attribute-value-changes'
[13.02.2025 07:52.31.542] [LOG]   sending subscriber notification to: HTTP://127.0.0.1:4013/v1/regard-attribute-value-change with content: {"MicroWaveDeviceInventory-1-2:attribute-value-changed-notification":{"counter":4,"timestamp":"2025-02-13T07:52:30.0+00:00","attribute-name":"atpc-is-on","new-value":"true"}} - debugId: 'f55f9206-4d85-4da7-8ca5-12f47af0306d'
      response: [Function: handleResponse],
      error: [Function: handleRequestError]
    },
    _eventsCount: 2,
    _maxListeners: undefined,
    _options: {
      maxRedirects: 21,
      maxBodyLength: 10485760,
      protocol: 'http:',
      path: '/v1/regard-attribute-value-change',
      method: 'POST',
      headers: [Object],
      agent: undefined,
      agents: [Object],
      auth: undefined,
      hostname: '127.0.0.1',
      port: '4013',
      nativeProtocols: [Object],
      pathname: '/v1/regard-attribute-value-change'
    },
    _ended: false,
    _ending: true,
    _redirectCount: 0,
    _redirects: [],
    _requestBodyLength: 174,
    _requestBodyBuffers: [ [Object] ],
    _onNativeResponse: [Function (anonymous)],
    _currentRequest: ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      maxRequestsOnConnectionReached: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      strictContentLength: false,
      _contentLength: 174,
      _hasBody: true,
      _trailer: '',
      finished: false,
      _headerSent: true,
      _closed: false,
      socket: [Socket],
      _header: 'POST /v1/regard-attribute-value-change HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Content-Type: application/json\r\n' +
        'x-correlator: e5F26ad2-dDF8-64Df-F178-7C3EEd0E810c\r\n' +
        'trace-indicator: 1\r\n' +
        'user: MicroWaveDeviceInventory\r\n' +
        'originator: MicroWaveDeviceInventory\r\n' +
        'customer-journey: unknown\r\n' +
        'operation-key: Operation key not yet provided.\r\n' +
        'User-Agent: axios/0.21.4\r\n' +
        'Content-Length: 174\r\n' +
        'Host: 127.0.0.1:4013\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: nop],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/v1/regard-attribute-value-change',
      _ended: false,
      res: null,
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: '127.0.0.1',
      protocol: 'http:',
      _redirectable: [Circular *1],
      [Symbol(kCapture)]: false,
      [Symbol(kBytesWritten)]: 0,
      [Symbol(kEndCalled)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype],
      [Symbol(kUniqueHeaders)]: null
    },
    _currentUrl: 'http://127.0.0.1:4013/v1/regard-attribute-value-change',
    [Symbol(kCapture)]: false
  },
  response: undefined,
  isAxiosError: true,
  toJSON: [Function: toJSON]
} error during subscriber-notification for /v1/notify-attribute-value-changes - debugId: 'f55f9206-4d85-4da7-8ca5-12f47af0306d'
[13.02.2025 07:52.31.679] [LOG]
 callback : POST http://172.28.127.18:3033/v1/record-service-request header :{"user":"NotificationProxy","originator":"MicroWaveDeviceInventory","x-correlator":"aa1dd54a-52da-4f3b-f65F-8EEECBE08f71","trace-indicator":"1","customer-journey":"unknown","operation-key":"Operation key not yet provided.","Content-Type":"application/json"}body :{"x-correlator":"6AbeD2f9-53CE-E9Bb-e4dD-b95dd035aC7D","trace-indicator":"1","user":"NotificationProxy","originator":"NotificationProxy","application-name":"MicroWaveDeviceInventory","release-number":"1.2.1","operation-name":"/core-model-1-4:network-control-domain=live/control-construct=513250006/logical-termination-point=LTP-MWPS-TTP-ODU-A/layer-protocol=LP-MWPS-TTP-ODU-A/air-interface-2-0:air-interface-pac/air-interface-configuration","response-code":200,"timestamp":"2025-02-13T07:52:31+00:00"}response code:204
[13.02.2025 07:52.31.724] [LOG]
 callback : POST http://172.28.127.18:3033/v1/record-service-request header :{"user":"NotificationProxy","originator":"MicroWaveDeviceInventory","x-correlator":"fb27Ff79-3eDC-B0C2-C90e-82BD3851B961","trace-indicator":"1","customer-journey":"unknown","operation-key":"Operation key not yet provided.","Content-Type":"application/json"}body :{"x-correlator":"6AbeD2f9-53CE-E9Bb-e4dD-b95dd035aC7D","trace-indicator":"1","user":"NotificationProxy","originator":"NotificationProxy","application-name":"MicroWaveDeviceInventory","release-number":"1.2.1","operation-name":"/v1/regard-device-attribute-value-change","response-code":204,"timestamp":"2025-02-13T07:52:31+00:00"}response code:204
[13.02.2025 07:52.31.739] [LOG]
 callback : POST http://172.28.127.18:3033/v1/record-service-request header :{"user":"MicroWaveDeviceInventory","originator":"MicroWaveDeviceInventory","x-correlator":"68fC0fdC-C00d-BAFb-eED9-F4F6989B411b","trace-indicator":"1","customer-journey":"unknown","operation-key":"Operation key not yet provided.","Content-Type":"application/json"}body :{"x-correlator":"e5F26ad2-dDF8-64Df-F178-7C3EEd0E810c","trace-indicator":"1","user":"MicroWaveDeviceInventory","originator":"MicroWaveDeviceInventory","application-name":"MicroWaveDeviceInventory","release-number":"1.2.1","operation-name":"/v1/notify-attribute-value-changes","response-code":500,"timestamp":"2025-02-13T07:52:31+00:00"}response code:204
GET /core-model-1-4:network-control-domain=cache/control-construct=CO18301?fields=forwarding-domain%28uuid%3Blayer-protocol-name%3Bmac-fd-1-0%3Amac-fd-pac%28mac-fd-status%28mac-address-cur%29%29%29 200 232.938 ms - 1253
[13.02.2025 07:52.38.024] [LOG]
 callback : POST http://172.28.127.18:3033/v1/record-service-request header :{"user":"MacAddressTableRecorder","originator":"MicroWaveDeviceInventory","x-correlator":"d0aBF8C3-f454-AbaE-9E9A-EEe8C35bDCfF","trace-indicator":"1","customer-journey":"unknown","operation-key":"Operation key not yet provided.","Content-Type":"application/json"}body :{"x-correlator":"7AC9DBFE-EbAb-a6B7-872a-cFbeFfDF1bbc","trace-indicator":"1","user":"MacAddressTableRecorder","originator":"MacAddressTableRecorder","application-name":"MicroWaveDeviceInventory","release-number":"1.2.1","operation-name":"/core-model-1-4:network-control-domain=cache/control-construct=CO18301?fields=forwarding-domain%28uuid%3Blayer-protocol-name%3Bmac-fd-1-0%3Amac-fd-pac%28mac-fd-status%28mac-address-cur%29%29%29","response-code":200,"timestamp":"2025-02-13T07:52:37+00:00"}response code:204
[13.02.2025 07:52.38.770] [LOG]
 callback : http://172.28.127.19:8181/rests/data/network-topology:network-topology/topology=topology-netconf/node=CO03045/yang-ext:mount/core-model-1-4:control-construct header :{"Authorization":"Basic bXdkaV90ZXN0YTptd2RpQDEyMw=="}body :""response code:200
[13.02.2025 07:52.38.811] [INFO]  Result is:  updated
[13.02.2025 07:52.38.821] [LOG]   [Resp. Time: 69580ms] Response from element CO03045. Dropped from S-W. Add element CO17798 in S-W and send request...
[13.02.2025 07:52.38.907] [LOG]
 callback : POST http://172.28.127.18:3033/v1/record-service-request header :{"user":"MicroWaveDeviceInventory","originator":"MicroWaveDeviceInventory","x-correlator":"3BbbD7E7-C77A-8BFc-7534-799bfabEa4b0","trace-indicator":"1","customer-journey":"unknown","operation-key":"Operation key not yet provided.","Content-Type":"application/json"}body :{"x-correlator":"9E8EFbb2-F07b-8465-bc43-648c285FCF7B","trace-indicator":"1","user":"MicroWaveDeviceInventory","originator":"MicroWaveDeviceInventory","application-name":"MicroWaveDeviceInventory","release-number":"1.2.1","operation-name":"/core-model-1-4:network-control-domain=live/control-construct=CO03045","response-code":200,"timestamp":"2025-02-13T07:52:38+00:00"}response code:204
GET /core-model-1-4:network-control-domain=cache/control-construct=CO18301/logical-termination-point=CO18301+LTP-MACINTERFACE-TTP-7-6/ltp-augment-1-0:ltp-augment-pac?fields=original-ltp-name 200 205.121 ms - 87
GET /core-model-1-4:network-control-domain=cache/control-construct=CO18301/logical-termination-point=CO18301+LTP-MACINTERFACE-TTP-3-1/ltp-augment-1-0:ltp-augment-pac?fields=original-ltp-name 200 193.690 ms - 86
GET /core-model-1-4:network-control-domain=cache/control-construct=CO06254?fields=forwarding-domain%28uuid%3Blayer-protocol-name%3Bmac-fd-1-0%3Amac-fd-pac%28mac-fd-status%28mac-address-cur%29%29%29 200 100.226 ms - 461

[13.02.2025 07:52.58.318] [LOG]   sending subscriber notification to: HTTP://127.0.0.1:4013/v1/regard-attribute-value-change with content: {"MicroWaveDeviceInventory-1-2:attribute-value-changed-notification":{"counter":4,"timestamp":"2025-02-13T07:52:30.0+00:00","attribute-name":"atpc-is-on","new-value":"true"}} - debugId: '50d715d0-8b5d-4433-b8f5-d0c7ddb7f8e1'
[13.02.2025 07:52.58.320] [ERROR] Error: connect ECONNREFUSED 127.0.0.1:4013
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 4013,
  config: {
    url: 'HTTP://127.0.0.1:4013/v1/regard-attribute-value-change',
    method: 'post',
    data: '{"MicroWaveDeviceInventory-1-2:attribute-value-changed-notification":{"counter":4,"timestamp":"2025-02-13T07:52:30.0+00:00","attribute-name":"atpc-is-on","new-value":"true"}}',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      'x-correlator': 'dD6D9c57-988d-78f6-E96b-B31fdCa747BE',
      'trace-indicator': '1',
      user: 'MicroWaveDeviceInventory',
      originator: 'MicroWaveDeviceInventory',
      'customer-journey': 'unknown',
      'operation-key': 'Operation key not yet provided.',
      'User-Agent': 'axios/0.21.4',
      'Content-Length': 174
    },
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    adapter: [Function: httpAdapter],
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    validateStatus: [Function: validateStatus],
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    }
  },
  request: <ref *1> Writable {
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: true,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      buffered: [],
      bufferedIndex: 0,
      allBuffers: true,
      allNoop: true,
      pendingcb: 0,
      constructed: true,
      prefinished: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: true,
      errored: null,
      closed: false,
      closeEmitted: false,
      [Symbol(kOnFinished)]: []
    },
    _events: [Object: null prototype] {
      response: [Function: handleResponse],
      error: [Function: handleRequestError]
    },
    _eventsCount: 2,
    _maxListeners: undefined,
    _options: {
      maxRedirects: 21,
      maxBodyLength: 10485760,
      protocol: 'http:',
      path: '/v1/regard-attribute-value-change',
      method: 'POST',
      headers: [Object],
      agent: undefined,
      agents: [Object],
      auth: undefined,
      hostname: '127.0.0.1',
      port: '4013',
      nativeProtocols: [Object],
      pathname: '/v1/regard-attribute-value-change'
    },
    _ended: false,
    _ending: true,
    _redirectCount: 0,
    _redirects: [],
    _requestBodyLength: 174,
    _requestBodyBuffers: [ [Object] ],
    _onNativeResponse: [Function (anonymous)],
    _currentRequest: ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      maxRequestsOnConnectionReached: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      strictContentLength: false,
      _contentLength: 174,
      _hasBody: true,
      _trailer: '',
      finished: false,
      _headerSent: true,
      _closed: false,
      socket: [Socket],
      _header: 'POST /v1/regard-attribute-value-change HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Content-Type: application/json\r\n' +
        'x-correlator: dD6D9c57-988d-78f6-E96b-B31fdCa747BE\r\n' +
        'trace-indicator: 1\r\n' +
        'user: MicroWaveDeviceInventory\r\n' +
        'originator: MicroWaveDeviceInventory\r\n' +
        'customer-journey: unknown\r\n' +
        'operation-key: Operation key not yet provided.\r\n' +
        'User-Agent: axios/0.21.4\r\n' +
        'Content-Length: 174\r\n' +
        'Host: 127.0.0.1:4013\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: nop],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/v1/regard-attribute-value-change',
      _ended: false,
      res: null,
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: '127.0.0.1',
      protocol: 'http:',
      _redirectable: [Circular *1],
      [Symbol(kCapture)]: false,
      [Symbol(kBytesWritten)]: 0,
      [Symbol(kEndCalled)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype],
      [Symbol(kUniqueHeaders)]: null
    },
    _currentUrl: 'http://127.0.0.1:4013/v1/regard-attribute-value-change',
    [Symbol(kCapture)]: false
  },
  response: undefined,
  isAxiosError: true,
  toJSON: [Function: toJSON]
} error during subscriber-notification for /v1/notify-attribute-value-changes - debugId: '50d715d0-8b5d-4433-b8f5-d0c7ddb7f8e1'
GET /core-model-1-4:network-control-domain=cache/control-construct=513559991B?fields=forwarding-domain%28uuid%3Blayer-protocol-name%3Bmac-fd-1-0%3Amac-fd-pac%28mac-fd-status%28mac-address-cur%29%29%29 200 177.165 ms - 941
[13.02.2025 07:52.58.590] [LOG]
```

--- a/server/service/individualServices/cacheUpdateBuilder.js
+++ b/server/service/individualServices/cacheUpdateBuilder.js
@@ -170,7 +170,7 @@ function assignValueToJson(json, path, nuovoJSON, filters) {
             if (nuovoJSON != null) {
               let objectKey = Object.keys(nuovoJSON)[0];
               let newJSON = nuovoJSON[objectKey][0];
-              oggetto[nomeArray][indice] = newJSON;
+              oggetto[chiave] = newJSON;
             }
           }
         } else {

--- a/server/service/individualServices/cacheUpdateBuilder.js
+++ b/server/service/individualServices/cacheUpdateBuilder.js
@@ -2,6 +2,7 @@ const createHttpError = require("http-errors");
 
 exports.cacheUpdateBuilder = function (url, originalJSON, toInsert, filters) {
   const urlParts = url.split("?fields=");
+  //TODO @latta-siae better check filters
   const myFields = urlParts[1];
   // Analyze URL to extract segments
   const urlSegments = urlParts[0].split('/').filter(segment => segment.trim() !== '');
@@ -21,27 +22,26 @@ exports.cacheUpdateBuilder = function (url, originalJSON, toInsert, filters) {
       continue;
     }
 
-
     const [key, value] = segment.split('=');
 
     // Verify if the property exists in the current JSON
     if (currentJSON.hasOwnProperty(key)) {
       if (Array.isArray(currentJSON[key])) {
         // If the field is an array try to find the elemnt with the corrispondent UUID
-        const uuidDaCercare = decodeURIComponent(value);
-        const indexDaSostituire = currentJSON[key].findIndex(item =>
-          (item.uuid && item.uuid === uuidDaCercare) ||
-          (item['local-id'] && item['local-id'] === uuidDaCercare)
+        const uuidToFind = decodeURIComponent(value);
+        const indexToChange = currentJSON[key].findIndex(item =>
+          (item.uuid && item.uuid === uuidToFind) ||
+          (item['local-id'] && item['local-id'] === uuidToFind)
         );
-        if (indexDaSostituire !== -1) {
+        if (indexToChange !== -1) {
           // Substitute only the element into the array with the new JSON
           if (currentJSON[key]) {
             if (i == 1) {
-              lastKey = "[" + objectKey + "]" + "." + key + "[" + indexDaSostituire + "]";
+              lastKey = "[" + objectKey + "]" + "." + key + "[" + indexToChange + "]";
             } else {
-              lastKey = lastKey + "." + key + "[" + indexDaSostituire + "]";
+              lastKey = lastKey + "." + key + "[" + indexToChange + "]";
             }
-            currentJSON = currentJSON[key][indexDaSostituire];
+            currentJSON = currentJSON[key][indexToChange];
           } else {
             console.warn(`Field "${key}" not found in cache`);
             break;
@@ -52,7 +52,7 @@ exports.cacheUpdateBuilder = function (url, originalJSON, toInsert, filters) {
           lastKey = lastKey + "." + key;
           //throw new createHttpError.NotFound(`Field "${key}"="${value}" not found in cache`);
 
-          console.warn(`No elements found with UUID: ${uuidDaCercare}`);
+          console.warn(`No elements found with UUID: ${uuidToFind}`);
           break;
         }
       } else {
@@ -70,11 +70,13 @@ exports.cacheUpdateBuilder = function (url, originalJSON, toInsert, filters) {
     }
     i += 1;
   }
+
   if (lastKey == null) {
     lastKey = "[" + objectKey + "]";
   }
+
   // Verify if exists a last key and substitute it with the new JSON
-  if (lastKey) {
+  if (lastKey) { // I think now is unuseful
     //console.log(originalJSON[lastKey])
 
     assignValueToJson(originalJSON, lastKey, toInsert, myFields);
@@ -82,9 +84,10 @@ exports.cacheUpdateBuilder = function (url, originalJSON, toInsert, filters) {
 };
 
 
-function assignValueToJson(json, percorso, nuovoJSON, filters) {
+// What does this function does?
+function assignValueToJson(json, path, nuovoJSON, filters) {
 
-  const chiavi = percorso.split('.');
+  const pathKeys = path.split('.');
 
   let oggetto = json;
   let Filters = false;
@@ -92,9 +95,9 @@ function assignValueToJson(json, percorso, nuovoJSON, filters) {
     Filters = true;
   }
   let nomeArray = "";
-  for (let i = 0; i < chiavi.length; i++) {
+  for (let i = 0; i < pathKeys.length; i++) {
     if (i == 0) {
-      const chiave = chiavi[i];
+      const chiave = pathKeys[i];
       const parentesiQuadraApertaIndex = chiave.indexOf('[');
       const parentesiQuadraChiusaIndex = chiave.indexOf(']');
       const nomeArray = chiave.substring(1, parentesiQuadraChiusaIndex);
@@ -108,7 +111,7 @@ function assignValueToJson(json, percorso, nuovoJSON, filters) {
         // oggetto = oggetto[nomeArray];
       }
       // If the key doesn't contain square brackets get the object value
-      if (i === chiavi.length - 1) {
+      if (i === pathKeys.length - 1) {
         // If this is the last key in the path, assign the new value
         if (Filters) {
           let objectKey = Object.keys(nuovoJSON)[0];
@@ -126,15 +129,16 @@ function assignValueToJson(json, percorso, nuovoJSON, filters) {
         }
       }
     } else {
-      const chiave = chiavi[i];
+      const chiave = pathKeys[i];
       const parentesiQuadraApertaIndex = chiave.indexOf('[');
       const parentesiQuadraChiusaIndex = chiave.indexOf(']');
 
+      // This happen when 
       if (parentesiQuadraApertaIndex !== -1 && parentesiQuadraChiusaIndex !== -1) {
         nomeArray = chiave.substring(0, parentesiQuadraApertaIndex);
         const indice = parseInt(chiave.substring(parentesiQuadraApertaIndex + 1, parentesiQuadraChiusaIndex), 10);
 
-        if (i === chiavi.length - 1) {
+        if (i === pathKeys.length - 1) {
           // If this is the last key in the path, assign the new value
           if (Filters) {
             let objectKey = Object.keys(nuovoJSON)[0];
@@ -156,7 +160,7 @@ function assignValueToJson(json, percorso, nuovoJSON, filters) {
         }
       } else {
         // If the key doesn't contain square brackets get the objet value
-        if (i === chiavi.length - 1) {
+        if (i === pathKeys.length - 1) {
           // Se questa è l'ultima chiave nel percorso, assegna il nuovo valore
           if (Filters) {
             let objectKey = Object.keys(nuovoJSON)[0];

